### PR TITLE
Add topic_message_retention_duration property to pubsub_topics module

### PR DESCRIPTION
**Pull request by victoriayexxx**
This will enable us to increase the hl7v2 topic retention to 31 days. (Currently using default of 7 days.)

### CODE REVIEW

File: examples/tfengine/resources_only.hcl [hunk 1]
```diff
@@ -106,6 +106,14 @@ template "resources" {
           member = "group:example-data-viewers@example.com"
         }]
       }]
+      topic_message_retention_duration = [{
```
Comment (fbeevikm) [inline examples/tfengine/resources_only.hcl]: This should generate the tf files here https://github.com/GoogleCloudPlatform/solution-acceleration-toolkit/tree/master/examples/tfengine/generated

Comment (fbeevikm) [reply examples/tfengine/resources_only.hcl]: https://github.com/GoogleCloudPlatform/solution-acceleration-toolkit/blob/8ae9b6385f25151d26cf8e64d844d02796e779fe/examples/tfengine/modules/team.hcl#L484

Add the config here.

**Review by yeweidaniel — APPROVED** [review]
/gcbrun

**Review by fbeevikm — CHANGES_REQUESTED** [review]

File: examples/tfengine/resources_only.hcl (hunk 1 above)
Comment (victoriayexxx) [reply examples/tfengine/resources_only.hcl]: > This should generate the tf files here https://github.com/GoogleCloudPlatform/solution-acceleration-toolkit/tree/master/examples/tfengine/generated

The new diff has the generated main.tf file.

**Review by fbeevikm — APPROVED** [review]

**Pull request merged**